kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/WorkflowRunnerViewModel.kt

```diff
@@ -104,3 +121,39 @@ internal class WorkflowRunnerViewModel<OutputT : Any>(
     val BUNDLE_KEY = WorkflowRunner::class.java.name + "-workflow"
   }
 }
+
+/**
+ * Attempts to use the `tracing-debugger` library to write a trace of the workflow session to the
+ * file.
+ */
+@UseExperimental(
+    VeryExperimentalWorkflow::class,
+    ExperimentalCoroutinesApi::class,
+    ExperimentalTime::class
+)
+private fun Flow<WorkflowDebugInfo>.tryStreamTraceToFile(
+  scope: CoroutineScope,
+  traceFile: File
+) {
+  Log.i("WorkflowRunner", "Streaming workflow trace to $traceFile")
+
+  // Ensure the caller has the tracing library in their classpath.
+  val traceEvents = try {
+    toTraceEvents()
+  } catch (e: NoClassDefFoundError) {
+    throw UnsupportedOperationException(
```

Comment:
Nice.

Reply:
The rewrite doesn't require this trickery, thank god.